Makefile

```diff
@@ -15,12 +15,16 @@ default: deploy
 # not really sure what to do with this target yet
 deploy-client: 
 
-deploy:
+deploy: deploy-service-scripts deploy-lib
+
+deploy-lib:
 	rm -rf $(SERVICE_DIR)/venv
 	virtualenv $(SERVICE_DIR)/venv
+	. $(SERVICE_DIR)/venv/bin/activate && pip install .
+
+deploy-service-scripts:
 	cp $(SCRIPTS_DIR)/start_service $(SCRIPTS_DIR)/stop_service $(SERVICE_DIR)/
 	rsync -avzP $(LBIN_DIR) $(SERVICE_DIR)/
```

Comment:
You probably do not need to rsync the bin directory from the git dir to the service dir, although it doesn't hurt.
Once you activate the virtualenv, its bin dir will become first in the path and will contain everything installed from bin.
It looks like you activate the virtualenv from start and stop service scripts, so you should be ok in that regard.